support-models/src/main/scala/com/gu/support/workers/PaymentMethods.scala

```diff
@@ -86,17 +86,32 @@ case class ClonedDirectDebitPaymentMethod(
   PaymentGateway: PaymentGateway = DirectDebitGateway
 ) extends PaymentMethod
 
+case class GatewayOption(name: String, value: String)
+case class GatewayOptionData(GatewayOption: List[GatewayOption])
+case class SepaPaymentMethod(
+  BankTransferAccountName: String,
+  BankTransferAccountNumber: String,
+  Email: String,
+  IPAddress: String,
+  GatewayOptionData: GatewayOptionData,
+  BankTransferType: String = "SEPA",
+  `Type`: String = "BankTransfer",
+  PaymentGateway: PaymentGateway = SepaGateway
+) extends PaymentMethod
+
 case class AmazonPayPaymentMethod(
   TokenId: String,
   Type: String = "CreditCardReferenceTransaction",  // This is how amazon pay works in zuora - as a credit card
   PaymentGateway: PaymentGateway
 ) extends PaymentMethod
 
 object PaymentMethod {
+  import io.circe.generic.auto._
```

Comment:
why auto rather than semiauto?  I'm sure there was a reason we use semiauto everywhere but I can't remember why at the moment.

Reply:
I added it for the `GatewayOptionData` type, which the `sepaPaymentMethodCodec` requires.
I've used `io.circe.generic.auto._` a lot in the past!

Reply:
Maybe we do semiauto in other places just to be more explicit about the implicits(!) so that it doesn't use the wrong/default codec by mistake.  But it's a classic tradeoff of extra code vs magic happens, I was just keeping it consistent.